### PR TITLE
soc: mcxa156: use soc hooks

### LINKED ISSUE #78386: z_arm_platform_init silently fails to be called following #62595
**Describe the bug**

- #62595 has renamed the function `z_arm_platform_init` `soc_reset_hook` (https://github.com/zephyrproject-rtos/zephyr/pull/62595/files#diff-da3175d625f6d13072c4f16348168e9b326d011de36fc65724821d3369774628R310) . I think this is breaking change in the kernel's API.
- In the Kconfig file at arch/arm/core/Kconfig
> The platform specific initialization code (z_arm_platform_init) is executed at the beginning of the startup code (__start)

This is not true: this function is never called by the kernel. Anyone who's `z_arm_platform_init` is implemented out-of-tree (or, in my case, rebased their changes on top of the current `HEAD` of `main`), will (potentially) only find out their function is not called if they've dug into this.

**To Reproduce**

See steps above.

**Expected behavior**

- I think this API change hasn't gone through the cycle documented here https://docs.zephyrproject.org/latest/develop/api/api_lifecycle.html#api-lifecycle
- This API change should be mentioned in the release notes or migration guide (I feel like this should have been a blocker to the PR being approved).
- https://docs.zephyrproject.org/latest/develop/api/api_lifecycle.html#deprecated seems to have a clearly defined approach to handling deprecation.

**Impact**

- Annoyance
- Unexpected failures at run time unless you've studied the code to find this error.
- Out-of-tree users are going to have a bad time.

### PR DESCRIPTION
**Pull request by nashif**
Use SoC hooks instead of legacy z_arm_platform_init.

Fixes #78386

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

### CODE REVIEW

**Review by henrikbrixandersen — DISMISSED** [review]

File: soc/nxp/mcx/mcxa/soc.c [hunk 1]
```diff
@@ -17,9 +17,9 @@
 #include <zephyr/init.h>
 #include <soc.h>
 
-#ifdef CONFIG_PLATFORM_SPECIFIC_INIT
+#ifdef CONFIG_SOC_RESET_HOOK
```
Comment (gmarull) [inline soc/nxp/mcx/mcxa/soc.c:20]: ifdef is useless, can be removed

**Review by henrikbrixandersen — DISMISSED** [review]

**Comment by nashif** [comment]
@mmahadevan108  @danieldegrasse  please review this one

File: soc/nxp/mcx/mcxa/soc.c [hunk 2]
```diff
@@ -17,11 +17,7 @@
 #include <zephyr/init.h>
 #include <soc.h>
 
-#ifdef CONFIG_PLATFORM_SPECIFIC_INIT
```
Comment (danieldegrasse) [inline soc/nxp/mcx/mcxa/soc.c:20]: I believe the `ifdef` here is not useless- we don't compile in the code implementing `SystemInit` unless `CONFIG_PLATFORM_SPECIFIC_INIT` is set: https://github.com/zephyrproject-rtos/hal_nxp/blob/master/mcux/hal_nxp.cmake#L101

So we might need a change to the HAL as well there, to use `CONFIG_SOC_RESET_HOOK`?

Comment (nashif) [reply soc/nxp/mcx/mcxa/soc.c:20]: ok, re-added it to be consistent with other usages, can be removed later everywhere if we decide it is not needed.

Comment (gmarull) [reply soc/nxp/mcx/mcxa/soc.c:20]: Doesn’t the soc always select the option?

**Review by gmarull — APPROVED** [review]

**Review by henrikbrixandersen — APPROVED** [review]

**Pull request merged**